awx/main/tests/data/inventory/plugins/azure_rm/files/azure_rm.yml

```diff
@@ -0,0 +1,20 @@
+default_host_filters: []
+hostvar_expressions:
+  computer_name: name
+  private_ip: private_ipv4_addresses | json_query("[0]")
+  provisioning_state: provisioning_state | title
+  type: resource_type
+keyed_groups:
```

Comment:
*In general*, for compatibility keyed_groups and vars, it would be good to have both behind a "compatibility with prior inventory scripts" flag that is set at a per-source level, which then causes this compatibility info to be templated into the plugin config.

Yes, it's more work, but it allows us to migrate off of them later in a sane way.

Reply:
That's where we're trying to go, yes. The question for me isn't "if", but "when".

What you're suggesting is going to result in substantial UI impact and rework. I kind of have an idea of how they might do this, but there could be some significant back-and-fourth with the API, and we would need to meet and start talking about design.

If you look at the foreman plugin, I made a PR to Ansible so that the credential secrets can be provided from environment variables. By doing this, I'm pushing all of the secret information out of the inventory file, so that I can (at a later point), surface this inventory file through the API, so that we can give it to the user in the way that you're talking about.

Reply:
How is this substantial UI work? It's a checkbox, on by default, that does nothing when using a script, and causes a "{{ ... }}" section to be dumped into the plugin configuration yml when set.

Reply:
Because there is no UI element for the inventory plugin configuration now, and if they added a box for this, they would need to hide / disable a bunch of other fields when it becomes active.